lib/graphql/metrics/analyzer.rb

```diff
@@ -93,37 +100,88 @@ def result
+      def arguments_for(node, definition)
+        # Arguments can raise execution errors within their `prepare` methods
+        # which aren't properly handled during analysis so we have to handle
+        # them ourselves safely and return `nil`.
+        query.arguments_for(node, definition)
+      rescue ::GraphQL::ExecutionError
+        nil
+      end
+
+      def extract_arguments(argument:, definition:, parent:, parent_input_object: nil)
         case argument
         when Array
           argument.each do |a|
-            extract_arguments(a, field_defn, parent_input_object)
+            extract_arguments(
+              argument: a,
+              definition: definition,
+              parent_input_object: parent_input_object,
+              parent: parent
+            )
           end
         when Hash
           argument.each_value do |a|
-            extract_arguments(a, field_defn, parent_input_object)
+            extract_arguments(
+              argument: a,
+              definition: definition,
+              parent_input_object: parent_input_object,
+              parent: parent)
           end
         when ::GraphQL::Execution::Interpreter::Arguments
           argument.each_value do |arg_val|
-            extract_arguments(arg_val, field_defn, parent_input_object)
+            extract_arguments(
+              argument: arg_val,
+              definition: definition,
+              parent_input_object: parent_input_object,
+              parent: parent
+          )
           end
         when ::GraphQL::Execution::Interpreter::ArgumentValue
-          extract_argument(argument, field_defn, parent_input_object)
-          extract_arguments(argument.value, field_defn, parent_input_object)
+          extract_argument(
+            value: argument,
+            definition: definition,
+            parent_input_object: parent_input_object,
+            parent: parent
+          )
+          extract_arguments(
+            argument:argument.value,
+            definition: definition,
+            parent_input_object: parent_input_object,
+            parent: parent
+          )
         when ::GraphQL::Schema::InputObject
           input_object_argument_values = argument.arguments.argument_values.values
           parent_input_object = input_object_argument_values.first&.definition&.owner
 
-          extract_arguments(input_object_argument_values, field_defn, parent_input_object)
+          extract_arguments(
+            argument: input_object_argument_values,
+            definition: definition,
+            parent_input_object: parent_input_object,
+            parent: parent
+          )
         end
       end
 
-      def extract_argument(value, field_defn, parent_input_object = nil)
+      def extract_argument(value:, definition:, parent_input_object:, parent:)
+        parent_type_name = if definition.is_a? GraphQL::Schema::Field
+          definition.owner.graphql_name
+        else
+          "__Directive"
```

Comment:
Can you constantize this?

Reply:
should we return here `GraphQL::Schema::Directive`?